### PR TITLE
Core: Move some variables from let to const.

### PR DESCRIPTION
**Pull request by DefinitelyMaybe**
Thank you to @marcofugaro for [this comment from a little bit ago](https://github.com/mrdoob/three.js/pull/19459#issuecomment-633887699). I modified the `package.json` file as per below and worked through the resulting linted errors.

```json
"eslintConfig": {
    ...
    "rules": {
      ...
      "prefer-const": ["error", {
        "destructuring": "any",
        "ignoreReadBeforeAssign": false
      }]
    }
  },
```

This PR has not included any change to the `package.json` file. Ya'll can fight that one out to your hearts content.

### CODE REVIEW

**Comment by DefinitelyMaybe** [comment]
interesting. When I run `npm test` on my machine, no errors came up however this is currently failing because something should still be a let and I've changed it to const. one moment.

**Comment by marcofugaro** [comment]
Might be useful for the future the set those eslint rules in the package.json, maybe in another PR.
Let's see what the maintainers say.

File: src/animation/AnimationMixer.js [hunk 1]
```diff
@@ -521,8 +521,8 @@ AnimationMixer.prototype = Object.assign( Object.create( EventDispatcher.prototy
 
 		const clipUuid = clipObject !== null ? clipObject.uuid : clip;
 
-		let actionsForClip = this._actionsByClip[ clipUuid ],
-			prototypeAction = null;
+		const actionsForClip = this._actionsByClip[ clipUuid ];
+		let	prototypeAction = null;
```
Comment (Mugen87) [inline src/animation/AnimationMixer.js:525]: No indentation is needed in such cases. It should be:
```js
const actionsForClip = this._actionsByClip[ clipUuid ];
let prototypeAction = null;
```

**Comment by Mugen87** [comment]
> Might be useful for the future the set those eslint rules in the package.json, maybe in another PR.
Let's see what the maintainers say.

I leave it up to @mrdoob if he wants to force `const` usage that way.

File: src/animation/AnimationMixer.js (hunk 1 above)
Comment (DefinitelyMaybe) [reply src/animation/AnimationMixer.js:525]: It's very strange that eslint isn't picking that up. I'll change and commit when I next get some time.

Comment (mrdoob) [reply src/animation/AnimationMixer.js:525]: Yes, there are many instances of these spaces.
Would be great if you can do a clean up pass.

File: src/animation/PropertyBinding.js [hunk 2]
```diff
@@ -445,12 +445,12 @@ Object.assign( PropertyBinding.prototype, { // prototype, continued
...
-			parsedPath = this.parsedPath,
+		let targetObject = this.node;
+		const parsedPath = this.parsedPath;
 
-			objectName = parsedPath.objectName,
-			propertyName = parsedPath.propertyName,
-			propertyIndex = parsedPath.propertyIndex;
+		const	objectName = parsedPath.objectName;
+		const	propertyName = parsedPath.propertyName;
+		let	propertyIndex = parsedPath.propertyIndex;
```
Comment (mrdoob) [inline src/animation/PropertyBinding.js:453]: Some more here.

File: src/extras/Earcut.js [hunk 3]
```diff
@@ -380,19 +380,18 @@ function eliminateHole( hole, outerNode ) {
...
 
-	let p = outerNode,
-		hx = hole.x,
-		hy = hole.y,
-		qx = - Infinity,
-		m;
+	let p = outerNode;
+	const	hx = hole.x;
+	const	hy = hole.y;
+	let	qx = - Infinity, m;
```
Comment (mrdoob) [inline src/extras/Earcut.js:386]: Some more here.

File: src/extras/Earcut.js [hunk 4]
```diff
@@ -328,8 +328,8 @@ function splitEarcut( start, triangles, dim, minX, minY, invSize ) {
 // link every hole into the outer loop, producing a single-ring polygon without holes
 function eliminateHoles( data, holeIndices, outerNode, dim ) {
 
-	let queue = [],
-		i, len, start, end, list;
+	const queue = [];
+	let	i, len, start, end, list;
```
Comment (mrdoob) [inline src/extras/Earcut.js:332]: One more here.

**Comment by mrdoob** [comment]
Adding that rule to eslint sounds good to me 👌

**Comment by DefinitelyMaybe** [comment]
thank you both. I'll open a quick PR for the eslint rule.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**